docs/pages/versions/unversioned/sdk/dataMigrationHelper.md

```diff
@@ -0,0 +1,57 @@
+---
+title: DataMigrationHelper
+---
+
+**`DataMigrationHelper`**  helps you to solve a problem with missing files after SDK upgrade on Android.
+The problem only occurs when one upgrades the ejected/standalone application from SDK 32 or below to SDK 33 or above.
+
+## API
+
+```js
+import { DataMigrationHelper } from 'expo';
+```
+
+### `DataMigrationHelper.getLegacyDocumentDirectoryAndroid(): string | null`
+
+This method returns the path to the legacy document directory which can be used by `expo-file-system`.
+
+> Returns null on all platforms except Android
+
+### `DataMigrationHelper.migrateFilesFromLegacyDirectoryAsync(): Promise<void>`
+
+```js
+
+export default class App extends React.Component {
+  state = {
+    movedFiles: false,
+  };
+
+  render() {
+    const { movedFiles } = this.state;
+
+    if (!movedFiles) {
+      return (
+        <AppLoading
+          startAsync={this._moveFiles}
+          onFinish={() => this.setState({ movedFiles: true })}
+          onError={console.warn}
+        />
+      );
+    }
+
+    return (
+      ...
+    );
+  }
+
+  async _moveFiles() {
+    await DataMigrationHelper.migrateFilesFromLegacyDirectoryAsync();
+  }
+  ...
+}
+
+```
+
+This method moves all files from the legacy directory to the current document directory. If the legacy directory is empty then it is a no-op. If the current document directory already contains a file with the same relative path as a file in the legacy directory then it will be overwritten.
```

Comment:
Would it be possible to add an option to preserve the file in the current directory if there is a naming conflict, rather than always overwriting with the file from the legacy directory? It seems like there is an important use case for that -- what if a developer already upgraded their SDK 32 app to SDK 33+ and then just created new data in the new location? For end users who have upgraded, the developer would NOT want to migrate their old data and overwrite the new data, but if there are some end users who have not yet upgraded to the SDK 33+ version, the developer in theory WOULD want to migrate the old data in that case.

This would essentially be adding the node `COPYFILE_EXCL` option here for the operation as a whole (but behind the scenes it would apply to each individual file in the directory). https://nodejs.org/api/fs.html#fs_fs_copyfile_src_dest_flags_callback